src/services/expenses.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+let expenses = [];
+
+function getEmptyExpenses() {
+  expenses = [];
+}
+
+function getAll() {
+  return expenses;
+}
+
+function filterAllByUserId(userId) {
+  return expenses.filter(
+    expense => expense.userId === +userId,
+  );
+}
+
+function filterAllByDate(from, to) {
+  const start = Date.parse(from);
+  const end = Date.parse(to);
+
+  return expenses.filter(expense => {
+    const currentDate = Date.parse(expense.spentAt);
+
+    return currentDate.valueOf() >= start.valueOf()
+      && currentDate.valueOf() <= end.valueOf();
+  });
+}
+
+function filterAllByCategory(category) {
+  return expenses.filter(
+    expense => expense.category === category,
+  );
+}
+
+function getById(expensesId) {
+  const foundExpense = expenses.find(expense => expense.id === +expensesId);
+
+  return foundExpense || null;
+}
+
+function create(body) {
+  const {
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note,
+  } = body;
```

Comment:
Jsut suggestion - in case of validation on comtroller you dont need here destructuring body, just put ...body to new object 🔥